odoo/fields.py

```diff
@@ -1318,6 +1318,11 @@ def __init__(self, string=Default, **kwargs):
             kwargs['translate'] = bool(kwargs['translate'])
         super(_String, self).__init__(string=string, **kwargs)
 
+    def _setup_attrs(self, model, name):
+        super(_String, self)._setup_attrs(model, name)
+        if callable(self.translate):
+            self.prefetch = False
```

Comment:
What would happen if someone adds a field such as:

```python
myfield = fields.Char(
  translate=translation_method,
  prefetch=True,
)
```

Would the field be prefetchable or not?

Reply:
I still think we should force it. What are the reasons for wanting to prefetch it?